include/oneapi/dpl/internal/function.h

```diff
@@ -90,6 +91,123 @@ get_access(const Policy& policy, T* ptr)
     assert(sycl::get_pointer_type(ptr, policy.queue().get_context()) == sycl::usm::alloc::shared);
     return ptr;
 }
+
+template <typename ValueType, typename Policy, typename Iterator>
+ValueType
+get_data_at(const Policy& p, Iterator i, size_t index)
+{
+    return get_access<sycl::access::mode::read>(p, i)[index];
```

Comment:
`get_access` will cause the whole buffer to be transferred to the host, while we only need a single item. I think it could be improved.

Reply:
Fixed, please check.

Reply:
Fixed. Please review get_buffer_for and it's usage.